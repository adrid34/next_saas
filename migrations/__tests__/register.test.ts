import { withRootDb } from "./test_helpers";
import { snapshotSafe } from "./test_helpers";

test("can register user", () =>
  withRootDb(async client => {
    const result = await client.query(
      "SELECT * FROM app_private.link_or_register_user($1, $2, $3, $4, $5)",
      [
        null,
        "facebook",
        "123456",
        JSON.stringify({ email: "a@b.c", firstName: "A", lastName: "B" }),
        JSON.stringify({}),
      ]
    );
    const [user] = result.rows;
    expect(user).not.toBeNull();
    expect(snapshotSafe(user)).toMatchSnapshot();
  }));
