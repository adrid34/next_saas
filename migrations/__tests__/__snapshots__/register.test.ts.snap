// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can register user 1`] = `
Object {
  "avatar_url": null,
  "created_at": "[DATE]",
  "id": "[ID]",
  "is_admin": true,
  "is_verified": true,
  "name": null,
  "updated_at": "[DATE]",
  "username": "user",
}
`;
